twisted/internet/base.py

```diff
@@ -601,18 +602,33 @@ def sigInt(self, *args):
         """Handle a SIGINT interrupt.
         """
         log.msg("Received SIGINT, shutting down.")
+        try:
+            import signal
```

Comment:
we need a comment to describe why we need this local import... I guess that this is due to Windows...

but we also need a comment to explain  this can not be implemented in the posix base reactor